### PR TITLE
crowdstrike: Return empty events array when no resources in alert, host.

### PR DESCRIPTION
**Pull request by kcreddy**
## Proposed commit message
Return empty `events` array when no `resources` in `alert` and `host` data-streams.

When there are no `resources` in first API call, current CEL code returns `state`.
But this `state` doesn't have `events` inside it. As per [CEL docs](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-cel.html#input-state-cel:~:text=The%20events%20field%20must%20be%20present), the `events` field 
is necessary. Without this, the following errors appear and lead to restarting of input.

```
{"log.level":"error","@timestamp":"2024-08-21T06:26:37.252Z","message":"unexpected missing events array from evaluation","component":{"binary":"filebeat","dataset":"elastic_agent.filebeat","id":"cel-default","type":"cel"},"log":{"source":"cel-default"},"input_source":"http://svc-crowdstrike-host:8090","input_url":"http://svc-crowdstrike-host:8090","ecs.version":"1.6.0","log.logger":"input.cel","log.origin":{"file.line":350,"file.name":"cel/input.go","function":"github.com/elastic/beats/v7/x-pack/filebeat/input/cel.input.run.func1"},"service.name":"filebeat","id":"cel-crowdstrike.host-20697e27-0353-4b36-8c17-18f5ea89ae90","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2024-08-21T06:26:37.252Z","message":"unregistering","component":{"binary":"filebeat","dataset":"elastic_agent.filebeat","id":"cel-default","type":"cel"},"log":{"source":"cel-default"},"input_type":"cel","key":"cel-crowdstrike_host-20697e27-0353-4b36-8c17-18f5ea89ae90::http://svc-crowdstrike-host:8090","ecs.version":"1.6.0","log.logger":"metric_registry","log.origin":{"file.line":70,"file.name":"inputmon/input.go","function":"github.com/elastic/beats/v7/libbeat/monitoring/inputmon.NewInputRegistry.func1"},"service.name":"filebeat","id":"cel-crowdstrike.host-20697e27-0353-4b36-8c17-18f5ea89ae90::http://svc-crowdstrike-host:8090","uuid":"4ee4a37e-1dde-445d-9f17-2429c435d848","ecs.version":"1.6.0"}
{"log.level":"error","@timestamp":"2024-08-21T06:26:37.252Z","message":"Input 'cel' failed with: input.go:130: input cel-crowdstrike.host-20697e27-0353-4b36-8c17-18f5ea89ae90 failed (id=cel-crowdstrike.host-20697e27-0353-4b36-8c17-18f5ea89ae90)\n\tunexpected type returned for evaluation events: <nil>","component":{"binary":"filebeat","dataset":"elastic_agent.filebeat","id":"cel-default","type":"cel"},"log":{"source":"cel-default"},"log.logger":"input.cel","log.origin":{"file.line":132,"file.name":"compat/compat.go","function":"github.com/elastic/beats/v7/filebeat/input/v2/compat.(*runner).Start.func1"},"service.name":"filebeat","id":"cel-crowdstrike.host-20697e27-0353-4b36-8c17-18f5ea89ae90","ecs.version":"1.6.0","ecs.version":"1.6.0"}
```
 

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

By modifying `"resources": []` in the `config-alert.yml` and `config-host.yml`:

**Before:** (example with `host`)
`eval "$(elastic-package stack shellinit)" && elastic-package test system --generate  -v --data-streams=host`
```
{"log.level":"error","@timestamp":"2024-08-21T06:26:37.252Z","message":"unexpected missing events array from evaluation","component":{"binary":"filebeat","dataset":"elastic_agent.filebeat","id":"cel-default","type":"cel"},"log":{"source":"cel-default"},"input_source":"http://svc-crowdstrike-host:8090","input_url":"http://svc-crowdstrike-host:8090","ecs.version":"1.6.0","log.logger":"input.cel","log.origin":{"file.line":350,"file.name":"cel/input.go","function":"github.com/elastic/beats/v7/x-pack/filebeat/input/cel.input.run.func1"},"service.name":"filebeat","id":"cel-crowdstrike.host-20697e27-0353-4b36-8c17-18f5ea89ae90","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2024-08-21T06:26:37.252Z","message":"unregistering","component":{"binary":"filebeat","dataset":"elastic_agent.filebeat","id":"cel-default","type":"cel"},"log":{"source":"cel-default"},"input_type":"cel","key":"cel-crowdstrike_host-20697e27-0353-4b36-8c17-18f5ea89ae90::http://svc-crowdstrike-host:8090","ecs.version":"1.6.0","log.logger":"metric_registry","log.origin":{"file.line":70,"file.name":"inputmon/input.go","function":"github.com/elastic/beats/v7/libbeat/monitoring/inputmon.NewInputRegistry.func1"},"service.name":"filebeat","id":"cel-crowdstrike.host-20697e27-0353-4b36-8c17-18f5ea89ae90::http://svc-crowdstrike-host:8090","uuid":"4ee4a37e-1dde-445d-9f17-2429c435d848","ecs.version":"1.6.0"}
{"log.level":"error","@timestamp":"2024-08-21T06:26:37.252Z","message":"Input 'cel' failed with: input.go:130: input cel-crowdstrike.host-20697e27-0353-4b36-8c17-18f5ea89ae90 failed (id=cel-crowdstrike.host-20697e27-0353-4b36-8c17-18f5ea89ae90)\n\tunexpected type returned for evaluation events: <nil>","component":{"binary":"filebeat","dataset":"elastic_agent.filebeat","id":"cel-default","type":"cel"},"log":{"source":"cel-default"},"log.logger":"input.cel","log.origin":{"file.line":132,"file.name":"compat/compat.go","function":"github.com/elastic/beats/v7/filebeat/input/v2/compat.(*runner).Start.func1"},"service.name":"filebeat","id":"cel-crowdstrike.host-20697e27-0353-4b36-8c17-18f5ea89ae90","ecs.version":"1.6.0","ecs.version":"1.6.0"}
```

**After:**
No error messages in agent logs.

`eval "$(elastic-package stack shellinit)" && elastic-package test system --generate  -v --data-streams=host,alert`

```
--- Test results for package: crowdstrike - START ---
╭─────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE     │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├─────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ crowdstrike │ alert       │ system    │ common    │ PASS   │ 37.831208584s │
│ crowdstrike │ host        │ system    │ common    │ PASS   │ 36.294177167s │
╰─────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: crowdstrike - END   ---
Done
```

### CODE REVIEW

File: packages/crowdstrike/data_stream/alert/agent/stream/cel.yml.hbs [hunk 1]
```diff
@@ -75,7 +75,7 @@ program: |
       }
     )
   ).as(state, state.with(
-    !has(state.resources) ? state : // Exit early due to GET failure or no resources to collect.
+    !has(state.resources) ? {"events": []} : // Exit early due to GET failure or no resources to collect.
```
Comment (chrisberkhout) [inline packages/crowdstrike/data_stream/alert/agent/stream/cel.yml.hbs:78]: That changes a couple of other things:
- it doesn't return the calculated `want_more` and `offset` (maybe not important)
- it doesn't return the error event in the case of a GET failure (important)

I think the change should instead be to insert `"events": [],` below the `?"resources"` line.

Same for both data streams.

**Review by chrisberkhout — CHANGES_REQUESTED** [review]

File: packages/crowdstrike/data_stream/alert/agent/stream/cel.yml.hbs (hunk 1 above)
Comment (kcreddy) [reply packages/crowdstrike/data_stream/alert/agent/stream/cel.yml.hbs]: Nice catch. I updated the CEL code with your suggestion and re-ran the system tests. No error messages in the logs, so its working.

**Review by andrewkroh — COMMENTED** [review]
> When there are no resources in first API call, current CEL code returns state

Can we mock this is our test scenario? Like have it return no resources in a response before it returns resources.

**Comment by kcreddy** [comment]
> Can we mock this in our system test scenario? Like have it return no resources in a response before it returns resources.

I was unable to mock the `no resources` scenario at the beginning, but at the end. i.e., first it returns resources and subsequent calls returns no resources. Commit for this change: [eb72037](https://github.com/elastic/integrations/pull/10831/commits/eb720373654491948e04be7828f89b85d7c765b2).

The issue I had with making it at the beginning was that I am unable to increment `offset` because `resources.size()` is 0. Whatever `offset` that I start with, I end up with same value, thus unable to move to next page. https://github.com/elastic/integrations/blob/main/packages/crowdstrike/data_stream/alert/agent/stream/cel.yml.hbs#L55-L58

**Review by andrewkroh — APPROVED** [review]

**Review by chrisberkhout — APPROVED** [review]

**Pull request merged**